projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/inlineassembly/LLVMInlineAssemblyTest.java

```diff
@@ -29,54 +29,73 @@
  */
 package com.oracle.truffle.llvm.test.inlineassembly;
 
+import static org.junit.Assert.assertEquals;
+
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
 
-import org.junit.Assert;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import com.oracle.truffle.api.source.Source;
 import com.oracle.truffle.api.vm.PolyglotEngine;
 import com.oracle.truffle.api.vm.PolyglotEngine.Builder;
+import com.oracle.truffle.llvm.runtime.LLVMLogger;
+import com.oracle.truffle.llvm.runtime.options.LLVMBaseOptionFacade;
 import com.oracle.truffle.llvm.test.LLVMPaths;
-import com.oracle.truffle.llvm.tools.Clang;
-import com.oracle.truffle.llvm.tools.Clang.ClangOptions;
-import com.oracle.truffle.llvm.tools.Opt;
-import com.oracle.truffle.llvm.tools.Opt.OptOptions;
-import com.oracle.truffle.llvm.tools.Opt.OptOptions.Pass;
+import com.oracle.truffle.llvm.test.LLVMSuiteTestCaseGenerator;
+import com.oracle.truffle.llvm.test.TestCaseFiles;
+import com.oracle.truffle.llvm.test.TestSuiteBase;
+import com.oracle.truffle.llvm.test.spec.SpecificationEntry;
+import com.oracle.truffle.llvm.test.spec.SpecificationFileReader;
 
-@SuppressWarnings({"static-method"})
-public final class LLVMInlineAssemblyTest {
+@RunWith(Parameterized.class)
+public class LLVMInlineAssemblyTest extends TestSuiteBase {
 
-    private static final String PATH = LLVMPaths.LOCAL_TESTS + "/../inlineassemblytests";
+    private final File bitCodeFile;
+    private final File expectedFile;
+    private final File originalFile;
 
-    @Test
-    public void test001() {
-        String file = "inlineassembly001";
-        Assert.assertEquals(42, run(file));
+    public LLVMInlineAssemblyTest(TestCaseFiles testCase) {
+        this.expectedFile = testCase.getExpectedResult();
+        this.bitCodeFile = testCase.getBitCodeFile();
+        this.originalFile = testCase.getOriginalFile();
     }
 
-    @Test
-    public void test002() {
-        String file = "inlineassembly002";
-        Assert.assertEquals(42, run(file));
+    @Parameterized.Parameters
+    public static List<TestCaseFiles[]> getTestCases() {
+        List<String> testCaseFileNames = new ArrayList<>();
+        for (File file : LLVMPaths.INLINEASSEMBLY_TESTS.listFiles()) {
+            testCaseFileNames.add(file.getName());
+        }
+        List<SpecificationEntry> testCaseFileSpecList = SpecificationFileReader.getFiles(testCaseFileNames, LLVMPaths.INLINEASSEMBLY_TESTS);
+        return collectIncludedFiles(testCaseFileSpecList, new LLVMSuiteTestCaseGenerator(LLVMBaseOptionFacade.testBinaryParser()));
     }
 
-    private static int run(String fileName) {
+    @Test
+    public void test() {
```

Comment:
Could you implement the test suite similar to `NWCCTestSuite` and get the reference result by compiling the test cases with GCC or Clang and then compare the executable's results with the result that Sulong produces? Then you do not have to create the `.reference_output` files by hand.